src/pyFAI/multi_geometry.py

```diff
@@ -342,3 +343,493 @@ def reset(self, collect_garbage=True):
+
+
+    def integrate2d_fiber(self, lst_data,
+                          npt_ip=1000, unit_ip=None, ip_range=None,
+                          npt_oop=1000, unit_oop=None, oop_range=None,
+                          sample_orientation=None,
+                          correctSolidAngle=True,
+                          lst_mask=None, dummy=None, delta_dummy=None,
+                          lst_variance=None,
+                          polarization_factor=None, dark=None, lst_flat=None,
+                          method=("no", "histogram", "cython"),
+                          normalization_factor=1.0, **kwargs):
+        """Performs 2D azimuthal integration of multiples frames, one for each geometry,
+        It wraps the method integrate2d_fiber of pyFAI.integrator.fiber.FiberIntegrator
+
+        :param lst_data: list of numpy array
+        :param int npt_ip: number of points to be used along the in-plane axis
+        :param pyFAI.units.UnitFiber/str unit_ip: unit to describe the in-plane axis. If not provided, it takes qip_nm^-1
+        :param list ip_range: The lower and upper range of the in-plane unit. If not provided, range is simply (data.min(), data.max()). Values outside the range are ignored. Optional.
+        :param int npt_oop: number of points to be used along the out-of-plane axis
+        :param pyFAI.units.UnitFiber/str unit_oop: unit to describe the out-of-plane axis. If not provided, it takes qoop_nm^-1
+        :param list oop_range: The lower and upper range of the out-of-plane unit. If not provided, range is simply (data.min(), data.max()). Values outside the range are ignored. Optional.
+        :param int sample_orientation: 1-4, four different orientation of the fiber axis regarding the detector main axis, from 1 to 4 is +90º
+        :param bool correctSolidAngle: correct for solid angle of each pixel if True
+        :param lst_mask: numpy.Array or list of numpy.array which mask the lst_data.
+        :param float dummy: value for dead/masked pixels
+        :param float delta_dummy: precision for dummy value
+        :param lst_variance: list of array containing the variance of the data. If not available, no error propagation is done
+        :type lst_variance: list of ndarray
+        :param float polarization_factor: polarization factor between -1 (vertical) and +1 (horizontal).
+                * 0 for circular polarization or random,
+                * None for no correction,
+                * True for using the former correction
+        :param ndarray dark: dark noise image
+        :param lst_flat: numpy.Array or list of numpy.array which flat the lst_data.
+        :param IntegrationMethod method: IntegrationMethod instance or 3-tuple with (splitting, algorithm, implementation)
+        :param float normalization_factor: Value of a normalization monitor
+        :return: regrouped intensity and unit arrays
+        :rtype: Integrate2dResult
+        """
+        if "npt_horizontal" in kwargs:
```

Comment:
Since this block of code is duplicated twice, it would make sense to have it in a (private) method. One may need to update the locals dict ...